build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/leafTask.ts

```diff
@@ -685,6 +685,14 @@ export abstract class LeafWithGlobInputOutputDoneFileTask extends LeafWithFileSt
 	 */
 	protected abstract getOutputGlobs(): Promise<readonly string[]>;
 
+	/**
+	 * @returns If the lock file should be included as input files for this task.
```

Comment:
I can't think of a case where a task would want to depend on the lockfile for output... but it also seems a little incomplete if I can only use this for input. Maybe the setting could be like the toggle that changes how input/output globs treat gitignore? Not a big deal.

Reply:
The purpose of depending on the lockfile by default is that would detect and put the tools and dependent libraries as "input" that causes the build.   Not sure it really makes sense for it to be the output of the task.   If the tool really modifies the lockfile, then it better add it manually.

Current input vs output doesn't make much difference, but I like the separation for the user to consider separately, and we might use it for output caching in the future.